lib/mllparty/connection_hub/connection_hub.ex

```diff
@@ -53,17 +54,18 @@ defmodule MLLParty.ConnectionHub do
   Sends a message to the client connection process for the given `ip` and `port`.
   """
   def send_message(ip, port, message, opts \\ []) do
-    pid =
+    {pid, is_new_connection} =
       case start_client(ip, port) do
         {:ok, pid} ->
-          pid
+          {pid, true}
 
         {:error, {:already_started, pid}} ->
-          pid
+          {pid, false}
       end
 
-    if opts[:wait_for_client_to_connect] do
-      wait_for_client_to_connect(pid)
+    wait_for_client_to_connect(pid)
+    if is_new_connection do
```

Comment:
@mfos239 if the monitor process crashes it won't be respawned unless `send_message()` is called again AND the client process also died (otherwise `start_client()` will return `:already_started` and `is_new_connection` will be false.

It's common to have processes that you want to start together and die together, in such cases the tool for the job is a [Supervisor](https://hexdocs.pm/elixir/1.14.3/Supervisor.html).

Note that a parent process will also die if it's _linked_ child process dies for any reason other than `:normal` (see: https://furlough.merecomplexities.com/elixir/otp/2021/06/08/what-happens-when-a-linked-process-dies.html)